src/main/java/org/sagebionetworks/web/client/widget/discussion/DiscussionThreadListWidget.java

```diff
@@ -78,21 +83,23 @@ public void onFailure(Throwable caught) {
 					public void onSuccess(PaginatedResults<DiscussionThreadBundle> result) {
 						for(DiscussionThreadBundle bundle: result.getResults()) {
 							DiscussionThreadWidget thread = ginInjector.createThreadWidget();
-							thread.configure(bundle, isCurrentUserModerator);
+							thread.configure(bundle, isCurrentUserModerator, new Callback(){
+
+								@Override
+								public void invoke() {
+									configure(forumId, isCurrentUserModerator, emptyListCallback);
+								}
+							});
 							view.addThread(thread.asWidget());
 						}
 						offset += LIMIT;
 						long numberOfThreads = result.getTotalNumberOfResults();
 						view.setLoadingVisible(false);
 						view.setLoadMoreButtonVisibility(offset < numberOfThreads);
-						if (numberOfThreads > 0) {
```

Comment:
I would expect the thread list widget to own the "empty" message UI (but I suspect you'll still need to call back to the parent so that visibility of other UI, like create thread, is changed).